### PR TITLE
Fixes password behavior on Vault module.

### PR DESCRIPTION
**Pull request by rjeffman**
This patch fixes handling of password and public_key files, parameter
validation depending on vault type, usage of `salt` attribute and data
retrieval.

Tests were updated to reflect the changes.

New example playbooks are added:

    playbooks/vault/vault-is-present-with-password-file.yml
    playbooks/vault/vault-is-present-with-public-key-file.yml
    playbooks/vault/retrive-data-asymmetric-vault.yml
    playbooks/vault/retrive-data-symmetric-vault.yml

### CODE REVIEW

File: plugins/modules/ipavault.py [hunk 1]
```diff
@@ -261,7 +272,8 @@ def find_vault(module, name, username, service, shared):
 
 
 def gen_args(description, username, service, shared, vault_type, salt,
-             public_key, vault_data):
+             password, password_file, public_key, public_key_file, vault_data,
```
Comment (seocam) [inline plugins/modules/ipavault.py:275]: I don't know if I'm missing something but it seems that password, password_file, vault_data and datafile_in are not used in the function. Do we actually need them here?

File: plugins/modules/ipavault.py [hunk 2]
```diff
@@ -555,6 +602,10 @@ def main():
                         commands.append([name, 'vault_remove_owner',
                                          owner_del_args])
 
+                    if vault_type == 'symmetric' \
+                       and 'ipavaultsalt' not in args:
+                        args['ipavaultsalt'] = os.urandom(32)
```
Comment (seocam) [inline plugins/modules/ipavault.py:607]: is it ok generate a random salt for symmetric? Are you able to retrieve this information afterwards without this salt?

Comment (rjeffman) [reply plugins/modules/ipavault.py:717]: Yes you can retrieve the `salt` (in the CLI, but not with Ansible). It is the same way the CLI command does this.

File: tests/vault/test_vault.yml [hunk 3]
```diff
@@ -132,19 +151,113 @@
...
+    failed_when: result.changed
+
   - name: Ensure asymmetric vault is present.
     ipavault:
       ipaadmin_password: SomeADMINpassword
       name: asymvault
       username: user01
-      description: A symmetric private vault.
+      description: An asymmetric private vault.
       vault_public_key:
```
[the remark this answers is not in the document]
Comment (rjeffman) [reply tests/vault/test_vault.yml:300]: I'll keep vault_password, and will change the other attributes.

File: tests/vault/test_vault.yml [hunk 4]
```diff
@@ -108,9 +128,8 @@
       ipaadmin_password: SomeADMINpassword
       name: symvault
       username: user01
-      vault_password: MyVaultPassword123
+      vault_password: SomeVAULTpassword
       vault_data: The world of π is half rounded.
```
[the remark this answers is not in the document]
Comment (rjeffman) [reply tests/vault/test_vault.yml:173]: I'll keep vault_password, and will change the other attributes.

File: playbooks/vault/data-archive-in-symmetric-vault.yml [hunk 5]
```diff
@@ -9,6 +9,6 @@
       ipaadmin_password: SomeADMINpassword
       name: symvault
       username: admin
-      vault_password: MyVaultPassword123
+      vault_password: SomeVAULTpassword
       vault_data: The world of π is half rounded.
```
[the remark this answers is not in the document]
Comment (rjeffman) [reply playbooks/vault/data-archive-in-symmetric-vault.yml:13]: I'll keep vault_password, and will change the other attributes.

File: tests/vault/test_vault.yml [hunk 6]
```diff
@@ -77,7 +98,7 @@
       ipaadmin_password: SomeADMINpassword
       name: symvault
       username: user01
-      vault_password: MyVaultPassword123
+      vault_password: SomeVAULTpassword
```
[the remark this answers is not in the document]
Comment (rjeffman) [reply tests/vault/test_vault.yml:142]: I'll keep vault_password, and will change the other attributes.

File: plugins/modules/ipavault.py (hunk 1 above)
Comment (rjeffman) [reply plugins/modules/ipavault.py:320]: Good catch. They were used during development, are not used anymore. Fixed.

File: playbooks/vault/vault-is-present-with-password-file.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+      dest: "{{ ansible_env.HOME }}/password.txt"
+      owner: "{{ ansible_user }}"
+      group: "{{ ansible_user }}"
+      mode: 0600
+  - ipavault:
+      ipaadmin_password: SomeADMINpassword
+      name: symvault
+      username: admin
+      vault_type: symmetric
+      vault_password_file: "{{ ansible_env.HOME }}/password.txt"
```
Comment (t-woerner) [inline playbooks/vault/vault-is-present-with-password-file.yml:19]: This parameter is not mentioned in the README.

Comment (rjeffman) [reply playbooks/vault/vault-is-present-with-password-file.yml:19]: thanks. will fix in another PR, as this one has been merged.

**Pull request merged**